VisualQnA/tests/test_compose_tgi_on_gaudi.sh

```diff
@@ -0,0 +1,222 @@
+    # Check if the microservices are running correctly.
+
+    # lvm microservice
+    validate_services \
+        "${ip_address}:9399/v1/lvm" \
+        "The image" \
+        "lvm" \
+        "lvm-gaudi-server" \
+        '{"image": "iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAFUlEQVR42mP8/5+hnoEIwDiqkL4KAcT9GO0U4BxoAAAAAElFTkSuQmCC", "prompt":"What is this?"}'
+}
+
+function validate_megaservice() {
+    sleep 15s
+    # Curl the Mega Service
+    validate_services \
+    "${ip_address}:8888/v1/visualqna" \
+    "The image" \
+    "visualqna-gaudi-backend-server" \
+    "visualqna-gaudi-backend-server" \
+    '{
+        "messages": [
+        {
+            "role": "user",
+            "content": [
+            {
+                "type": "text",
+                "text": "What'\''s in this image?"
+            },
+            {
+                "type": "image_url",
+                "image_url": {
+                "url": "https://www.ilankelman.org/stopsigns/australia.jpg"
+                }
+            }
+            ]
+        }
+        ],
+        "max_tokens": 300
+    }'
+
+    # test the megeservice via nginx
+    validate_services \
+    "${ip_address}:80/v1/visualqna" \
+    "The image" \
+    "visualqna-gaudi-nginx-server" \
+    "visualqna-gaudi-nginx-server" \
+    '{
+        "messages": [
+        {
+            "role": "user",
+            "content": [
+            {
+                "type": "text",
+                "text": "What'\''s in this image?"
+            },
+            {
+                "type": "image_url",
+                "image_url": {
+                "url": "https://www.ilankelman.org/stopsigns/australia.jpg"
+                }
+            }
+            ]
+        }
+        ],
+        "max_tokens": 300
+    }'
+}
+
+function validate_frontend() {
+    cd $WORKPATH/ui/svelte
+    local conda_env_name="OPEA_e2e"
+    export PATH=${HOME}/miniforge3/bin/:$PATH
+    if conda info --envs | grep -q "$conda_env_name"; then
+        echo "$conda_env_name exist!"
+    else
+        conda create -n ${conda_env_name} python=3.12 -y
+    fi
+    source activate ${conda_env_name}
+
+    sed -i "s/localhost/$ip_address/g" playwright.config.ts
+
+    conda install -c conda-forge nodejs=22.6.0 -y
+    npm install && npm ci && npx playwright install --with-deps
+    node -v && npm -v && pip list
+
+    exit_status=0
+    npx playwright test || exit_status=$?
+
+    if [ $exit_status -ne 0 ]; then
+        echo "[TEST INFO]: ---------frontend test failed---------"
+        exit $exit_status
+    else
+        echo "[TEST INFO]: ---------frontend test passed---------"
+    fi
+}
+
+function stop_docker() {
+    cd $WORKPATH/docker_compose/intel/hpu/gaudi
+    docker compose stop && docker compose rm -f
+}
+
+function main() {
+
+    stop_docker
+
+    if [[ "$IMAGE_REPO" == "opea" ]]; then build_docker_images; fi
+    start_services
+
+    validate_microservices
+    validate_megaservice
+    # validate_frontend
```

Comment:
can we open the UI test?

Reply:
will open an issue and assign @WenjiaoYue

Reply:
https://github.com/opea-project/GenAIExamples/issues/1667